app/components/edit/EditAddress.jsx

```diff
@@ -17,79 +18,142 @@ class EditAddress extends Component {
     });
   }
 
+  handleNoLocationChange(e) {
+    this.setState((state, props) => {
+      let newAddr;
+      if (state.noLocation) {
+        newAddr = this.props.address;
+      } else {
+        newAddr = {
+          "address_1": "",
+          "address_2": "",
+          "address_3": "",
+          "address_4": "",
+          "city": "",
+          "country": "",
+          "postal_code": "",
+          "state_province": ""
+        };
+      }
+      return {noLocation: !state.noLocation, address: newAddr};
+    }, () => {
+      this.props.updateAddress(this.state.address);
+    });
+  }
+
   render() {
     const address = this.props.address || {};
+    return (
+      <AddressForm handleAddressChange={this.handleAddressChange}
+      handleNoLocationChange={this.handleNoLocationChange} 
+      noLocation={this.state.noLocation} address={address} />
+    );
+  }
+}
 
+const hasNoLocation = (address) => {
+  let someFieldExistsOrNewAddress = typeof address === "undefined" || address.address_1 != "" ||
+    address.address_2 != "" || address.address_1 != "" || address.address_3 != "" || 
+    address.address_4 != "" || address.city != "" || address.postal_code != "" ||
+    address.state_province != "" || address.country != "";
+  if (someFieldExistsOrNewAddress) {
+    return false;
+  }
+  return true;
+};
+
+const AddressForm = ({noLocation, handleNoLocationChange, handleAddressChange, address}) => {
+  if (noLocation) {
     return (
       <li key="address" className="edit--section--list--item">
         <label htmlFor="address">Address</label>
+        <input
+          type="checkbox"
+          className="input-checkbox"
+          checked={noLocation}
+          onChange={handleNoLocationChange} 
+        />
+        <div onClick={handleNoLocationChange}>No Physical Location</div>
+      </li>
+    );
+  } else {
+    return (
+      <li key="address" className="edit--section--list--item">
+        <label htmlFor="address">Address</label>
+        <input
+          type="checkbox"
+          className="input-checkbox"
+          checked={noLocation}
+          onChange={handleNoLocationChange} 
+        />
+        <div onClick={handleNoLocationChange}>No Physical Location</div>
```

Comment:
This seems to be an exact duplicate of the stuff above. Could you structure the `if` statement to make the common HTML always present and only conditionally hide the other stuff? You can do [conditional rendering](https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator) of components directly in JSX instead of in a normal JavaScript `if` statement.

Reply:
doing conditional rendering on all adjacent input tags except one (the checkbox) caused the error "adjacent jsx elements must be wrapped in an enclosing tag". the alternative is to wrap all the conditional inputs into one <div>. i'll make that change and you can check it out